### PR TITLE
API Wrapper

### PR DESCRIPTION
**Pull request by rusheel**

### CODE REVIEW

File: api_wrapper.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
+var request = require('request')
+var config = require('./config')
+
+function createAccount(name, type, cb) {
+  var options = {
+  uri: 'http://api.reimaginebanking.com/customers/580ba86e360f81f104544d31/accounts?key='+config.CapitalOneKey,
```
Comment (timmui) [inline api_wrapper.js]: move user to const at top

Comment (AdrianMachado) [inline api_wrapper.js]: Make the user a constant at the top of the file

File: api_wrapper.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  else {
+    cb("Could not create account. Please try again")
+  }
+})};
+
+exports.createAccount = createAccount;
+
+
+function getAccounts(type,cb){
+  request("http://api.reimaginebanking.com/customers/580ba86e360f81f104544d31/accounts?key="+config.CapitalOneKey
```
Comment (timmui) [inline api_wrapper.js]: Same

File: api_wrapper.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  json:
+    {
+      "type": type,
+      "nickname": name,
+      "rewards": 0,
+      "balance": 0
+        }
+  }
+request(options, function (error, response, body) {
+  if (!error && response.statusCode >= 200) {
```
Comment (AdrianMachado) [inline api_wrapper.js]: Is 404 not a thing?

File: config.js [hunk 4]
```diff
@@ -16,4 +16,5 @@ module.exports = {
     LuisSubscriptionKey: process.env.LUIS_SUBSCRIPTION_KEY,
     LuisAppId: process.env.LUIS_APP_ID,
     ComputerVisionSubscriptionKey: process.env.COMPUTER_VISION_SUBSCRIPTION_KEY,
+    CapitalOneKey: process.env.CAPITALONE_KEY
```
Comment (timmui) [inline config.js]: Trailing comma pls

File: api_wrapper.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  request("http://api.reimaginebanking.com/customers/580ba86e360f81f104544d31/accounts?key="+config.CapitalOneKey
+  , function (error, response, body) {
+    if (!error && response.statusCode == 200) {
+      var jsonResponse = JSON.parse(body);
+      returnString = '';
+      for (var i = 0; i < jsonResponse.length; i++) {
+        if (jsonResponse[i].type != type) continue;
+        var comma = ',';
+        if (i==jsonResponse.length-1) comma = '';
+        returnString = returnString + jsonResponse[i].type + " (" + jsonResponse[i].nickname + ") "
```
Comment (timmui) [inline api_wrapper.js]: single quotes pls

File: api_wrapper.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+        + '$' + jsonResponse[i].balance.toFixed(2) + comma + "\n\n";
```
Comment (timmui) [inline api_wrapper.js]: single quotes pls

File: api_wrapper.js [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+
+function getAccounts(type,cb){
+  request("http://api.reimaginebanking.com/customers/580ba86e360f81f104544d31/accounts?key="+config.CapitalOneKey
+  , function (error, response, body) {
+    if (!error && response.statusCode == 200) {
+      var jsonResponse = JSON.parse(body);
+      returnString = '';
+      for (var i = 0; i < jsonResponse.length; i++) {
+        if (jsonResponse[i].type != type) continue;
```
Comment (AdrianMachado) [inline api_wrapper.js]: Maybe use !== and check if it's the same behaviour

File: api_wrapper.js [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+      "type": type,
+      "nickname": name,
+      "rewards": 0,
+      "balance": 0
+        }
+  }
+request(options, function (error, response, body) {
+  if (!error && response.statusCode >= 200) {
+    console.log(body)
+    cb("Account Successfully Created")
```
Comment (timmui) [inline api_wrapper.js]: single quotes pls

File: api_wrapper.js [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+  }
+  else {
+    cb("Could not create account. Please try again")
```
Comment (timmui) [inline api_wrapper.js]: single quotes pls

**Review by timmui — CHANGES_REQUESTED** [review]
Pls fix comments

File: api_wrapper.js [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+    }})};
+
+  exports.getAccounts = getAccounts;
+
+  function findAtms(cb) {
+    request('http://api.reimaginebanking.com/atms?lat=38.9072&lng=-77.1753&rad=5&key='+config.CapitalOneKey, function (error, response, body) {
+      if (!error && response.statusCode == 200) {
+        var jsonResponse = JSON.parse(body).data;
+        console.log("The nearest ATM's are: ")
+        str = '';
```
Comment (AdrianMachado) [inline api_wrapper.js]: Put var

File: api_wrapper.js (hunk 8 above)
Comment (timmui) [reply api_wrapper.js]: also, put return before cb pls

File: config.js (hunk 4 above)
Comment (AdrianMachado) [inline config.js]: Rename CAPITAL_ONE_KEY

**Review by AdrianMachado — CHANGES_REQUESTED** [review]

File: config.js (hunk 4 above)
Comment (rusheel) [reply config.js]: i miss my linter

File: api_wrapper.js [hunk 11]
```diff
@@ -0,0 +1,62 @@
+var request = require('request')
+var config = require('./config')
+var username = '580ba86e360f81f104544d31';
```
Comment (timmui) [inline api_wrapper.js]: const not var

**Review by AdrianMachado — APPROVED** [review]

**Comment by AdrianMachado** [comment]
LGTM

File: api_wrapper.js [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+  exports.getAccounts = getAccounts;
+
+  function findAtms(cb) {
+    request('http://api.reimaginebanking.com/atms?lat=38.9072&lng=-77.1753&rad=5&key='+config.CapitalOneKey, function (error, response, body) {
+      if (!error && response.statusCode == 200) {
+        var jsonResponse = JSON.parse(body).data;
+        console.log('The nearest ATMs are: ')
+        var str = '';
+        for (var i = 0; i < 4; i++) {
+          str = str + (i+1) + ') ' + jsonResponse[i].name + ' , ';
```
Comment (timmui) [inline api_wrapper.js]: `str +=` instead

File: app.js [hunk 13]
```diff
@@ -55,10 +55,19 @@ dialog.matches('CreateAccount', [
...
         if (results.response) {
-            session.send(`Ok, opening a ${results.response.entity} account.`);
+          session.userData = results.response.entity;
+          builder.Prompts.text(session, 'Enter a name for your new ' + results.response.entity + ' account');
         } else {
             session.send('Something went wrong.');
         }
+    },
+    (session, results) => {
+      console.log(results.response)
```
Comment (timmui) [inline app.js]: remove console log

File: app.js [hunk 14]
```diff
@@ -82,7 +92,8 @@ dialog.matches('ViewAccount', [
 
 dialog.matches('FindATM', [
     (session, args, next) => {
-        session.send('Here are ATMs.');
+      session.send("Here are the nearest ATM's")
```
Comment (timmui) [inline app.js]: Single quotes pls. Use \ to escape '

**Review by timmui — APPROVED** [review]
Fix if you want

**Pull request merged**